Tentacle/Content.swift

```diff
@@ -0,0 +1,136 @@
+//
+//  Content.swift
+//  Tentacle
+//
+//  Created by Romain Pouclet on 2016-11-28.
+//  Copyright © 2016 Matt Diephouse. All rights reserved.
+//
+
+import Foundation
+import Argo
+import Curry
+import Runes
+
+public enum Content {
+    public struct File: CustomStringConvertible {
+        public enum ContentType: String {
+            case file
+            case dir
+            case symlink
+            case submodule
```

Comment:
I think we should make the URL of the submodule visible if possible. Also, we should have a test for a directory that contains a submodule because of [this behavior](https://developer.github.com/v3/repos/contents/#response-if-content-is-a-directory). I'm curious if we can detect that case by looking for `submodule_git_url` or something (instead of switching on the actual type that the API returns).